java/arcs/core/data/expression/Builders.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+@file:Suppress("UNCHECKED_CAST")
+
+/** Extension, infix, and operator overload functions for constructing Expression DSL. */
+package arcs.core.data.expression
+
+/** Constructs a [Expression.NumberLiteralExpression] */
+fun Double.asExpr() = Expression.NumberLiteralExpression(this)
+
+/** Constructs a [Expression.NumberLiteralExpression] */
+fun Int.asExpr() = Expression.NumberLiteralExpression(this)
+
+/** Constructs a [Expression.TextLiteralExpression] */
+fun String.asExpr() = Expression.TextLiteralExpression(this)
+
+/** Constructs a [Expression.BooleanLiteralExpression] */
+fun Boolean.asExpr() = Expression.BooleanLiteralExpression(this)
+
+/** Constructs a [Expression.Scope] for looking up query parameter references. */
+fun ParameterScope() = mutableMapOf<String, Any>().asScope()
+
+/** Constructs a [Expression.UnaryExpression] with [Expression.UnaryOp.Not]. */
+operator fun Expression<Boolean>.not() = Expression.UnaryExpression(Expression.UnaryOp.Not, this)
```

Comment:
I like this operators! 

A small nit: I would group all the Boolean ops separate from the Number ops.